go/test/endtoend/vtgate/tablet_healthcheck/reparent_test.go

```diff
@@ -0,0 +1,243 @@
+			SchemaSQL: schemaSQL,
+			VSchema:   vSchema,
+		}
+		clusterInstance.VtTabletExtraArgs = append(clusterInstance.VtTabletExtraArgs, []string{"--health_check_interval", "1s"}...)
+		err = clusterInstance.StartKeyspace(*keyspace, shards, 0, false)
+		if err != nil {
+			return 1
+		}
+
+		clusterInstance.VtGateExtraArgs = append(clusterInstance.VtGateExtraArgs, []string{"--tablet_refresh_interval", tabletRefreshInterval.String()}...)
+		err = clusterInstance.StartVtgate()
+		if err != nil {
+			return 1
+		}
+
+		vtParams = mysql.ConnParams{
+			Host: clusterInstance.Hostname,
+			Port: clusterInstance.VtgateMySQLPort,
+		}
+		return m.Run()
+	}()
+	os.Exit(exitCode)
+}
+
+// TestHealthCheckExternallyReparentNewTablet ensures that calling TabletExternallyReparented on a new tablet will switch the primary tablet
+// without having to wait for the tabletRefreshInterval.
+func TestHealthCheckExternallyReparentNewTablet(t *testing.T) {
+	ctx := context.Background()
+
+	// verify output of `show vitess_tablets` and `INSERT` statement
+	vtgateConn, err := mysql.Connect(ctx, &vtParams)
+	require.Nil(t, err)
+	defer vtgateConn.Close()
+
+	// add a new tablet
+	reparentTabletUID := 9999
+	reparentTabletType := "rdonly"
+	tablet := addTablet(t, reparentTabletUID, reparentTabletType)
+
+	// promote the new tablet to the primary
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("TabletExternallyReparented", tablet.Alias)
+	require.NoError(t, err)
+
+	// update the new primary tablet to be read-write
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("SetReadWrite", tablet.Alias)
+	require.NoError(t, err)
+
+	// wait for the vtgate to finish updating the new primary tablet
+	time.Sleep(1 * time.Second)
+
+	// verify that the vtgate will recognize the new primary tablet
+	qr, _ := vtgateConn.ExecuteFetch("show vitess_tablets", 100, true)
+	require.Equal(t, 3, len(qr.Rows), "wrong number of tablet records in healthcheck, expected %d but had %d", 3, len(qr.Rows))
+	require.Equal(t, "-80", qr.Rows[0][2].ToString())
+	require.Equal(t, "PRIMARY", qr.Rows[0][3].ToString())
+	require.Equal(t, "SERVING", qr.Rows[0][4].ToString())
+	require.Equal(t, tabletAlias(reparentTabletUID), qr.Rows[0][5].ToString())
+
+	// delete the old primary tablet
+	// This will ensure that the vtgate will experience the primary connection error if the switch would have to wait for the `tabletRefreshInterval`.
+	deleteTablet(t, clusterInstance.Keyspaces[0].Shards[0].Vttablets[0])
+
+	// verify that the vtgate will route the `INSERT` statement to the new primary tablet instead of the deleted tablet
+	qr, err = vtgateConn.ExecuteFetch("insert into customer(customer_id, email) values(2, 'dummy1')", 100, true) // -80
+	require.EqualValues(t, 1, qr.RowsAffected)
+	require.NoError(t, err)
+}
+
+func tabletAlias(tabletUID int) string {
+	return fmt.Sprintf("%s-%010d", cell, tabletUID)
+}
+
+func addTablet(t *testing.T, tabletUID int, tabletType string) *cluster.Vttablet {
```

Comment:
Is there anything unique about this function? I feel like we have other utilities we could use here instead.

Reply:
Can you tell me where I can find the common addTablet utility function? And, if it's not available, do you think it's a good idea to create a new utility function since I've referenced [the adjacent test file](https://github.com/vitessio/vitess/blob/8c3e37c28bec8f76e818402127e8d37addf77867/go/test/endtoend/vtgate/tablet_healthcheck_cache/correctness_test.go#L176) in the first place?